src/test/java/seedu/recipe/storage/jsonadapters/JsonAdaptedSubstitutionIngredientTest.java

```diff
@@ -0,0 +1,44 @@
+package seedu.recipe.storage.jsonadapters;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static seedu.recipe.testutil.Assert.assertThrows;
+import static seedu.recipe.testutil.TypicalRecipes.CACIO_E_PEPE;
+
+import java.util.Set;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.recipe.model.recipe.ingredient.Ingredient;
+
+public class JsonAdaptedSubstitutionIngredientTest {
+    private static final String VALID_INGREDIENT_NAME = "Sugar";
+    private static final String INVALID_INGREDIENT_NAME = "";
+
```

Comment:
Are there other invalid Ingredient Names?

Reply:
Can consider making them a static import from `Ingredient` class @alson001 @ZHTang29

Reply:
This should be done in model tests.